plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/wizard/resource/ResourcePayloadPage.java

```diff
@@ -0,0 +1,204 @@
+
+import com.openshift.restclient.OpenShiftException;
+
+/**
+ * @author Jeff Maury
+ *
+ */
+public class ResourcePayloadPage extends AbstractProjectPage<IResourcePayloadPageModel> {
+    
+    private static final IValidator URL_VALIDATOR =  new SimpleUrlStringValidator();
+
+    public ResourcePayloadPage(IWizard wizard, IResourcePayloadPageModel model) {
+        super(wizard, model,
+              "Select resource payload",
+              "Select the file or workspace file from where the resource will be created",
+              "resourcePayload");
+    }
+
+    @Override
+    protected void doCreateControls(Composite parent, DataBindingContext dbc) {
+        super.doCreateControls(parent, dbc);
+        createPayloadSourceControls(parent, dbc);
+    }
+
+    private void createPayloadSourceControls(Composite parent, DataBindingContext dbc) {
+        Group sourceGroup = new Group(parent, SWT.NONE);
+        sourceGroup.setText("Source");
+        GridDataFactory.fillDefaults()
+                .align(SWT.FILL, SWT.FILL)
+                .grab(true, true)
+                .span(3, 1)
+                .hint(SWT.DEFAULT, SWT.DEFAULT)
+                .applyTo(sourceGroup);
+        GridLayoutFactory.fillDefaults()
+                .numColumns(3)
+                .margins(10, 6)
+                .spacing(6, 6)
+                .applyTo(sourceGroup);
+        
+        createLocalSourceControls(dbc, sourceGroup);
+    }
+
+    private void createLocalSourceControls(DataBindingContext dbc, Group sourceGroup) {
+        Label label = new Label(sourceGroup, SWT.NONE);
+        GridDataFactory.fillDefaults().span(3, 1).applyTo(label);
+        label.setText("Enter a file path (workspace or local) or a full URL");
+        // local template file name
+        Text sourceText = new Text(sourceGroup, SWT.BORDER);
+        GridDataFactory.fillDefaults()
+                .align(SWT.FILL, SWT.CENTER).grab(true, false)
+                .applyTo(sourceText);
+        final IObservableValue source = WidgetProperties.text(SWT.Modify).observe(sourceText);
+        Binding binding = ValueBindingBuilder
+                .bind(source )
+                //.validatingBeforeSet(value->isFile(value.toString())?
+                //        ValidationStatus.ok(): 
+                //            ValidationStatus.error(value +" is not a file"))
+                .to(BeanProperties.value(
+                        IResourcePayloadPageModel.PROPERTY_SOURCE).observe(model))
+                .in(dbc);
+        MultiValidator validator = new MultiValidator() {
+            
+            @Override
+            protected IStatus validate() {
+                String sourceValue = (String) source.getValue();
+                return (boolean) !IResourcePayloadPageModel.URL_VALIDATOR.isValid(sourceValue) && !isFile(sourceValue)?ValidationStatus.error(sourceValue + " is not a file"):ValidationStatus.ok();
```

Comment:
I'd check for an empty sourceValue and treat this case separately since we dont want to show an error upon opening the dialog
![image](https://cloud.githubusercontent.com/assets/25126/14988458/2e701cf2-1154-11e6-9d2e-0be66af8e89f.png)